artemis-server/src/main/java/org/apache/activemq/artemis/core/deployers/impl/FileConfigurationParser.java

```diff
@@ -768,6 +772,63 @@ public void parseMainConfig(final Element e, final Configuration config) throws
       }
    }
 
+
+   private void parseJournalRetention(final Element e, final Configuration config) {
+      NodeList retention = e.getElementsByTagName("journal-retention");
+
+      if (retention.getLength() != 0) {
+         Element node = (Element) retention.item(0);
+
+         String storageLimitStr = getAttributeValue(node, "storage-limit");
+         long storageLimit;
+
+         if (storageLimitStr == null) {
+            storageLimit = -1;
+         } else {
+            storageLimit = ByteUtil.convertTextBytes(storageLimitStr.trim());
+         }
+         int period = getAttributeInteger(node, "period", -1, Validators.GT_ZERO);
+         String directory = getAttributeValue(node, "directory");
+         String unitStr = getAttributeValue(node, "unit");
+
+         if (unitStr == null) {
+            unitStr = "DAYS";
+         }
+
+         TimeUnit unit;
+
+         switch (unitStr) {
+            case "DAYS":
+               unit = TimeUnit.DAYS;
+               break;
+            case "HOURS":
+               unit = TimeUnit.HOURS;
+               break;
+            case "MINUTES":
```

Comment:
Here we support minutes but schema only has hours, days, seconds, i guess we want to add minutes to the schema / xsd